reactive/kubernetes_control_plane.py

```diff
@@ -1526,8 +1526,11 @@ def send_api_urls():
     kube_control.set_api_endpoints(kubernetes_control_plane.get_api_urls(endpoints))
 
 
-def has_external_cloud_provider():
-    return bool(hookenv.relations().get("external-cloud-provider"))
+def has_external_cloud_provider() -> bool:
+    has_xcp = bool(hookenv.relations().get("external-cloud-provider"))
+    if data_changed("has-xcp", has_xcp):
+        set_flag("external-cloud-provider.changed")
```

Comment:
On initial deploy, `data_changed` will be true, but this `e-c-p.changed` flag handler won't immediately reconfigure services since they won't be started yet.

On charm upgrade, the first call to this function will end up in this truthy `data_changed` and ultimately lead to a service reconfig, but that's ok because we want to ensure the `--cloud-provider=external` opt is set correctly on apiserver|cm|kubelet for all control-plane units going foward.